iquip/apps/builder.py

```diff
@@ -0,0 +1,419 @@
+        # widgets
+        self.checkBox = QCheckBox(self)
+        self.checkBox.setChecked(default)
+        # layout
+        self.layout.addWidget(self.checkBox)
+
+    def value(self) -> bool:
+        """Overridden.
+        
+        Returns the status of the checkBox.
+        """
+        return self.checkBox.isChecked()
+
+
+class _EnumerationEntry(_BaseEntry):
+    """Entry class for an enumeration value."""
+
+    def __init__(
+        self,
+        name: str,
+        choices: List[str],
+        default: Optional[str] = None,
+        parent: Optional[QWidget] = None
+    ):
+        """Extended.
+        
+        Args:
+            choices: The pre-defined candidates.
+            default: The default value. If it does not exist, it is set to the first candidate.
+        """
+        super().__init__(name, parent=parent)
+        # widgets
+        self.comboBox = QComboBox(self)
+        for choice in choices:
+            self.comboBox.addItem(choice)
+        if choices:
+            self.comboBox.setCurrentText(choices[0] if default is None else default)
+        # layout
+        self.layout.addWidget(self.comboBox)
+
+    def value(self) -> str:
+        """Overridden.
+        
+        Returns the value of the comboBox.
+        """
+        return self.comboBox.currentText()
+
+
+class _NumberEntry(_BaseEntry):
+    """Entry class for a number value.
+    
+    Attributes:
+        scale: The scale factor that actually applies.
+        type: The type of the value. If "int", value() returns an integer value.
+    """
+
+    def __init__(
+        self,
+        name: str,
+        unit: str,
+        scale: float,
+        step: float,
+        min: Optional[float],  # pylint: disable=redefined-builtin
+        max: Optional[float],  # pylint: disable=redefined-builtin
+        ndecimals: int,
+        type: str,  # pylint: disable=redefined-builtin
+        default: Optional[float] = None,
+        parent: Optional[QWidget] = None
+    ):  # pylint: disable=too-many-arguments
+        """Extended.
+        
+        Args:
+            unit: The unit of the value.
+            step: The step between values changed by the up and down button.
+            min: The minimum value. None for no minimum.
+            max: The maximum value. None for no maximum.
+            ndecimals: The maximum number of decimals.
+            default: The default value. If it does not exist, it is set to the min value.
+            scale, type: See the attributes section in _NumberEntry.
+        """
+        super().__init__(name, parent=parent)
+        self.scale = scale
+        self.type = type
+        # widgets
+        self.spinBox = QDoubleSpinBox(self)
+        self.spinBox.setSuffix(unit)
+        self.spinBox.setSingleStep(step / scale)
+        if min is not None:
+            self.spinBox.setMinimum(min / scale)
+        if max is not None:
+            self.spinBox.setMaximum(max / scale)
```

Comment:
If you see [`minimum`](https://doc.qt.io/qt-6/qdoublespinbox.html#minimum-prop) and [`maximum`](https://doc.qt.io/qt-6/qdoublespinbox.html#maximum-prop), they are not boundless by default. You have to set them with `float("inf")`, for example.